core/src/main/java/org/mule/runtime/core/internal/routing/IdempotentMessageValidator.java

```diff
@@ -72,11 +76,31 @@ public void initialise() throws InitialisationException {
           format("%s.%s.%s", muleContext.getConfiguration().getId(), getLocation().getRootContainerName(),
                  this.getClass().getName());
     }
+    setupObjectStore();
+  }
+
+  private void setupObjectStore() throws InitialisationException {
+    //Check if  OS was properly configured
+    if (store != null && privateStore != null) {
+      throw new InitialisationException(createStaticMessage("Ambiguous definition of object store, both reference and private were configured"),
+                                        this);
+    }
     if (store == null) {
-      this.store = createMessageIdStore();
+      if (privateStore == null) //If no object store was defined, create one
+      {
+        this.store = createMessageIdStore();
+      } else { //If object store was defined privatelly
+        this.store = privateStore;
+      }
+    }
+    initialiseIfNeeded(store, true, muleContext);
```

Comment:
if in the future, I decide to make the Extnesion OBject Store stoppable or disposable for whatever reason, that shouldn't mean that we have a leak just because you didn't do it here